cargo-manifest fails to build using Cargo from Fenix, but succeeds using rustup Cargo

To reproduce:
- Clone https://github.com/LukeMathWalker/cargo-manifest.git
- Checkout `f5e2ee275027058617f76777caf356a381277fc2`
- Run `cargo build`

The error occurs somewhere around `toml` and `indexmap`. The error does not appear when building `toml` on its own.

<details>

<summary>Error text (click to expand</summary>

```
   Compiling toml v0.5.11
error[E0107]: struct takes 3 generic arguments but 2 generic arguments were supplied
  --> /home/anselmschueler/.cargo/registry/src/index.crates.io-6f17d22bba15001f/toml-0.5.11/src/map.rs:39:22
   |
39 | type MapImpl<K, V> = IndexMap<K, V>;
   |                      ^^^^^^^^ -  - supplied 2 generic arguments
   |                      |
   |                      expected 3 generic arguments
   |
note: struct defined here, with 3 generic parameters: `K`, `V`, `S`
  --> /home/anselmschueler/.cargo/registry/src/index.crates.io-6f17d22bba15001f/indexmap-1.9.3/src/map.rs:76:12
   |
76 | pub struct IndexMap<K, V, S> {
   |            ^^^^^^^^ -  -  -
help: add missing generic argument
   |
39 | type MapImpl<K, V> = IndexMap<K, V, S>;
   |                                   +++

For more information about this error, try `rustc --explain E0107`.
error: could not compile `toml` (lib) due to previous error
```

</details>

--- a/lib/mk-toolchain.nix
+++ b/lib/mk-toolchain.nix
@@ -118,7 +118,8 @@ let
     components;
 
   toolchain' = toolchain // {
-    toolchain = combine "rust${suffix}-${date}" (attrValues toolchain);
+    toolchain = combine "rust${suffix}-${date}"
+      (attrValues (removeAttrs toolchain [ "rustc-dev" ]));
   } // optionalAttrs (toolchain ? rustc) {
     rustc = combine "rust${suffix}-with-std-${date}"
       (with toolchain; [ rustc rust-std ]) // {

